File: keep_alive.py
import requests

def keep_alive():
    url = "https://your-app-name.onrender.com/chat"
    payload = {"message": "ping"}
    try:
        response = requests.post(url, json=payload)
        print(f"Status Code: {response.status_code}, Response: {response.json()}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    keep_alive()

File: keep_alive.js
const https = require('https');

const data = JSON.stringify({
  message: 'ping'
});

const options = {
  hostname: 'xai-assistant-1.onrender.com',
  path: '/chat',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
};

const req = https.request(options, (res) => {
  console.log(`Status Code: ${res.statusCode}`);

  res.on('data', (d) => {
    process.stdout.write(d);
  });
});

req.on('error', (error) => {
  console.error('Error:', error);
});

req.write(data);
req.end();

console.log('Keep-alive ping sent (Node.js)');
